File: SlideWindow/3LengthOfLongestSubstring/lengthOfLongestSubstring.py
class Solution:
  def lengthOfLongestSubstring(self, s: str) -> int:
    left, res, size = 0, 0, len(s)
    set_ = set()
    for i in range(0, size):
      while s[i] in set_:
        set_.remove(s[left])
        left += 1
      res = max(res, i - left + 1)
      set_.add(s[i])
    return res
        

File: SlideWindow/3LengthOfLongestSubstring/lengthOfLongestSubstring.java
import java.util.HashSet;

class Solution {
  public int lengthOfLongestSubstring(String s) {
    int res = 0;
    HashSet<Character> set = new HashSet<>();
    int left = 0;
    for(int i = 0; i < s.length(); i++) {
      while(set.contains(s.charAt(i))){
        set.remove(s.charAt(left));
        left++;
      }
      res = Math.max(res, i - left + 1);
      set.add(s.charAt(i));
    }
    return res;
  }
}